### PR TITLE
Support product param

### LINKED ISSUE #110: The `product` field is missing in `CreatePlan`.
According to the API docs here https://stripe.com/docs/api/plans/create the `product` field is required.
Screen Shot 2020-01-27 at 4 09 18 PM

After looking through the docs & the source, it looks like that field is indeed missing on the `CreatePlan` struct.

### PR DESCRIPTION
**Pull request by stearnsc**
Closes #110 

Impls the solution discussed in https://github.com/wyyerd/stripe-rs/pull/111

### CODE REVIEW

**Comment by thedodd** [comment]
+1 looks good to me. Thanks for knocking this out!

File: src/params.rs [hunk 1]
```diff
@@ -278,6 +278,13 @@ impl<T> RangeQuery<T> {
     }
 }
 
+#[derive(Clone, Debug, Serialize)]
+#[serde(untagged)]
+pub enum IdOrCreate<'a, T> {
+    Id(&'a str),
+    Create(&'a T),
+}
+
```
Comment (thedodd) [inline src/params.rs]: Looks like we will need to make this public in the parent module as well. Look like the `params` module is private.

Comment (thedodd) [reply src/params.rs]: Yea, just needs to be added to the `pub use` block on `L81-83` in `src/lib.rs`.

Comment (stearnsc) [reply src/params.rs]: Ah yep, good catch

**Pull request merged**